### PR TITLE
Flav/serialize requested group ids

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

See https://github.com/dust-tt/dust/issues/8268.

With the recent addition of restricted/unrestricted spaces that starts creating a one to many relationship between spaces and groups. The current `groupIds` column has some limitations as it only performs a conjunction between all the group ids. Which does not work for unrestricted spaces that leverage the existing global group. This PR adds a new column on `conversations` and `agent_configurations` to serialize the `requestedGroupIds`. This new columns will serialized the requested group ids from the requested permissions preserving the disjunction of the group ids for a given space.

This is only the first step, that add the new column and enable the double write. Once the migration will be run, a second step will follow to start reading from it.

## Deploy Plan

- [x] Run SQL migration
- [x] Deploy `front`
- [x] Run the backfill
- [x] PR to switch reads
- [ ] PR to remove double write and drop previous column

### CODE REVIEW

**Comment by philipperolet** [comment]
:wave: Thanks! Would the colmn be added _in addition_ to groupIds? I had in mind that we would remove the `groupIds` column and directly add a full `acl` column with all the generality, is that was this is? in that case what about naming it `acl` directly?

**Comment by flvndvd** [comment]
> 👋 Thanks! Would the colmn be added _in addition_ to groupIds? I had in mind that we would remove the `groupIds` column and directly add a full `acl` column with all the generality, is that was this is? in that case what about naming it `acl` directly?

The main reason we don't want to store the `acl` for now is because permissions (like `read`, `write` are defined in the code not in the DB). Ultimately we know that at some point we will store the space Id to abstract all those permissions but for the time being we agreed to only store the group ids.

**Review by spolu — APPROVED** [review]
LGTM

File: front/lib/api/assistant/conversation.ts
Comment (spolu) [inline front/lib/api/assistant/conversation.ts:2091]: Are you sure about this one? Looks off

Comment (spolu) [reply front/lib/api/assistant/conversation.ts:2091]: If true then all newRequirements follow 
```
currentRequirements.some((currentReq) =>
      isEqual(sortBy(newReq), sortBy(currentReq))
    )
```
    
    Then below
 ```
  const missingRequirements = newRequirements.filter(
    (newReq) =>
      !currentRequirements.some((currentReq) =>
        isEqual(sortBy(newReq), sortBy(currentReq))
      )
  );
  ```
  
  would be empty?

File: front/lib/api/assistant/permissions.ts
Comment (spolu) [inline front/lib/api/assistant/permissions.ts:26]: Can you comment on this one to explain what it does precisely?

**Review by spolu — COMMENTED** [review]
(Added a few questions to address before merging)

File: front/lib/api/assistant/conversation.ts
Comment (flvndvd) [reply front/lib/api/assistant/conversation.ts:1409]: Yes exactly. See PR's description.

File: front/lib/api/assistant/conversation.ts
Comment (flvndvd) [reply front/lib/api/assistant/conversation.ts:2091]: Not sure to understand. Explaining the code here:

The code checks if all new requirements are already included in the conversation's existing requirements (using sorted arrays for comparison).
If `hasNewRequirements` is true, it means all new requirements are already present, so we can exit early. Otherwise, we compute the missing ones.

(will rename the variables to make it clearer!)

Anything that I missed?

File: front/lib/api/assistant/permissions.ts
Comment (flvndvd) [reply front/lib/api/assistant/permissions.ts:33]: Done

File: front/lib/api/assistant/conversation.ts
Comment (flvndvd) [reply front/lib/api/assistant/conversation.ts]: Dropping my tests here, if it helps:

```
>> newRequirements [
  [ 'group1', 'group2' ],
  [ 'group3', 'group4' ],
  [ 'group5', 'group6' ],
  [ 'group7', 'group8' ]
]
>> currentRequirements [ [ 'group1', 'group2' ] ]
>> hasNewRequirements false
>> missingRequirements [
  [ 'group3', 'group4' ],
  [ 'group5', 'group6' ],
  [ 'group7', 'group8' ]
]
```

Comment (spolu) [reply front/lib/api/assistant/conversation.ts]: Ok was confused by the name hasNewRequirements 👍 Should be named alreadyHasNewRequirements I think otherwise it's naturally understood the other way around.

Comment (flvndvd) [reply front/lib/api/assistant/conversation.ts]: Went with `areAllRequirementsPresent`.

**Review by spolu — APPROVED** [review]
LGTM modulo the RawAction stuff that should be cleaned up right?

**Pull request merged**
